cpp/src/io/parquet/page_string_utils.cuh

```diff
@@ -98,21 +100,40 @@ __device__ inline void block_excl_sum(size_type* arr, size_type length, size_typ
   }
 }
 
-template <int block_size, bool has_lists>
-__device__ inline void convert_small_string_lengths_to_offsets(page_state_s* s)
+/**
+ * @brief Converts string sizes to offsets if this is not a large string column. Otherwise,
+ * atomically update the initial string offset to be used during large string column construction
+ */
+template <int block_size>
```

Comment:
`has_lists` can't be a tparam anymore as it is not known at compile time when called from `page_delta_decode.cu`. Also, we are only using it minimally at L120